gpu/test/preconditioner/block_jacobi_kernels.cpp

```diff
@@ -107,6 +108,80 @@ class BlockJacobi : public ::testing::Test {
 };
 
 
+TEST_F(BlockJacobi, GpuFindNaturalBlocksEquivalentToRef)
+{
+    /* example matrix:
+        1   1
+        1   1
+        1       1
+        1       1
+     */
+    auto mtx = Mtx::create(ref);
+    mtx->read({4,
+               4,
+               {{0, 0, 1.0},
+                {0, 1, 1.0},
+                {1, 0, 1.0},
+                {1, 1, 1.0},
+                {2, 0, 1.0},
+                {2, 2, 1.0},
+                {3, 0, 1.0},
+                {3, 2, 1.0}}});
```

Comment:
Is this small matrix enough to verify that the GPU implementation is correct? Does it launch a kernel with multiple blocks to check there are no errors in thread ID computations?

Reply:
No, it is not. I see the need to add this. But I have to point out that we never do that. E.g. for none of the GPU solver kernel tests we use a vector size larger 512 (default thread block size).

Reply:
Actually you're right this is a problem. We didn't think of it before, we only tested so that we use more than a warp and with uncommon values.

Reply:
I created #41

Reply:
I honestly thought everyone was aware of this... this is why I suggested reference kernels in the first place, and made sure I test this in all my kernels. But I wasn't doing GPU solver kernels, and obviously didn't pay enough attention when reviewing the merge requests.

Reply:
Also, is there any reason for block size 512? Or is it just a random number someone came up with?

Reply:
512 is the default block size in the GPU kernels (e.g. for CGS CG FCG BICGSTAB...)